pedpy/data/geometry.py

```diff
@@ -66,3 +66,211 @@ def add_obstacle(self, obstacle: Polygon) -> None:
                 f"The obstacle {obstacle} is not inside the walkable area of "
                 "the geometry and thus will be ignored!"
             )
+
+
+###############################################################################
+# Measurement Area
+###############################################################################
+class MeasurementArea:
+    """Areas to study pedestrian dynamics.
+
+    A measurement area is defined as an area, which is convex, simple, and
+    covers a non-zero area.
+    """
+
+    _polygon: shapely.Polygon
+    _frozen = False
+
+    def __init__(self, coordinates: Any):
+        """Create a measurement area from the given input.
+
+        The measurement area may be a convex, simple area which covers a
+        non-zero area.
+
+        Args:
+            coordinates: A sequence of (x, y [,z]) numeric coordinate pairs, or
+                an array-like with shape (N, 2). Also, can be a sequence of
+                shapely.Point objects. Passing a wkt representation of a polygon
+                is also allowed.
+        """
+        try:
+            if isinstance(coordinates, shapely.Polygon):
+                self._polygon = coordinates
+            elif isinstance(coordinates, str):
+                self._polygon = shapely.from_wkt(coordinates)
+            else:
+                self._polygon = shapely.polygons(
+                    shapely.LinearRing(coordinates)
+                )
+        except Exception as exc:
+            raise ValueError(
+                f"Could not create measurement area from the given coordinates: {exc}."
+            ) from exc
+
+        if not isinstance(self._polygon, shapely.Polygon):
+            raise ValueError("Could not create a polygon from the given input.")
```

Comment:
Probably better handled in the cases above, here you lost context already and are not able to produce a good error message.